server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -129,7 +130,7 @@ private SearchHits buildSearchHits(SearchContext context, int[] docIdsToLoad, Pr
         StoredFieldLoader storedFieldLoader = profiler.storedFields(StoredFieldLoader.fromSpec(storedFieldsSpec));
         IdLoader idLoader = context.newIdLoader();
         boolean requiresSource = storedFieldsSpec.requiresSource();
-
+        final int[] accumulatedBytesInLeaf = new int[1];
```

Comment:
out of curiosity: why an array of ints with length 1, and what do we mean by "leaf" in the name?